v3/src/data-interactive/handlers/di-handler-utils.ts

```diff
@@ -7,16 +7,28 @@ import { v2NameTitleToV3Title } from "../../models/data/v2-model"
 import { ISharedCaseMetadata } from "../../models/shared/shared-case-metadata"
 import { getSharedCaseMetadataFromDataset } from "../../models/shared/shared-data-utils"
 import { hasOwnProperty } from "../../utilities/js-utils"
-import { DIAttribute, DICollection } from "../data-interactive-data-set-types"
+import { DIAttribute, DICategoryColorMap, DICollection } from "../data-interactive-data-set-types"
 import { convertValuesToAttributeSnapshot } from "../data-interactive-type-utils"
 
+function applyColormap(attributeId: string, colormap: DICategoryColorMap, metadata?: ISharedCaseMetadata) {
+  if (metadata) {
+    const categorySet = metadata.getCategorySet(attributeId)
+    Object.entries(colormap).forEach(([category, color]) => {
+      categorySet?.setColorForCategory(category, color)
+    })
+  }
+}
+
 export function createAttribute(value: DIAttribute, dataContext: IDataSet, collection?: ICollectionModel,
                                 metadata?: ISharedCaseMetadata) {
   const attributeSnapshot = convertValuesToAttributeSnapshot(value)
   if (attributeSnapshot) {
     const attribute = dataContext.addAttribute(attributeSnapshot, { collection: collection?.id })
     if (value.formula) attribute.formula?.setDisplayExpression(value.formula)
     metadata?.setIsHidden(attribute.id, !!value.hidden)
+    if (value.colormap) {
+      applyColormap(attribute.id, value.colormap as DICategoryColorMap, metadata)
```

Comment:
It shouldn't be necessary to use an `as` here. The problem is that `colormap` is typed as `any` in `ICodapV2Attribute`. I think that if you type that properly, you won't have to use `as` here or on line 70 below.